digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -946,6 +949,24 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
     }
   }
 
+  put("/node") {
+    time(logger, s"PUT request for /node") {
+      try {
+        val nodeInfo = parsedBody.extract[NodeExtractor]
+        val user = userProvider.getCurrentUser()
+        val node: Node = NodeConverter.toNode(nodeInfo, user)
+        nodesAndJunctionsService.addOrUpdateNode(node) match {
+          case Some(err) => Map("success" -> false, "errorMessage" -> err)
+          case None => Map("success" -> true)
+        }
+      } catch {
+        case e: Exception => {
+          logger.info("oops", e)
```

Comment:
we should use a better suggestive exception

Reply:
Actually, I've removed that now.
It was only needed for testing.